### PR TITLE
Kd_l2522_ag_macc_irr_mgmt

### PR DESCRIPTION
**Pull request by kdorheim**
#525

### CODE REVIEW

File: R/zchunk_L2522.ag_MACC_IRR_MGMT.R [hunk 1]
```diff
@@ -34,117 +33,164 @@ module_emissions_L2522.ag_MACC_IRR_MGMT_DISABLED <- function(command, ...) {
...
+
+
+    # Separate the MAC curves by SPP scenario.
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Agricultural MAC curve for SSP1
+    L2522.AgMAC_all_SSP_tech.change %>%
```
Comment (bpbond) [inline R/zchunk_L2522.ag_MACC_IRR_MGMT.R]: Perhaps making a little `filter_ssp` function that filters and drops `scenario`, and then calling it six times.

Comment (bpbond) [reply R/zchunk_L2522.ag_MACC_IRR_MGMT.R]: I.e.

```R
filter_spp <- function(x, ssp) {
  x %>%
       filter(scenario == ssp) %>%
       select(-scenario) 
}
```

File: R/zchunk_L2522.ag_MACC_IRR_MGMT.R [hunk 2]
```diff
@@ -34,117 +33,164 @@ module_emissions_L2522.ag_MACC_IRR_MGMT_DISABLED <- function(command, ...) {
...
+      L2522.AgMAC_all_SSP
+
+    # Add SSP scenarios to the the MAC animal data frame.
+    L252.MAC_an %>%
+      repeat_add_columns(scenario_list) ->
+      L2522.MAC_an_all_SSP
+
+
+    # Add the technology change to the agricultural and animal MAC curves by matching SSP
+    # scenario and mac.control (spplysector).
```
Comment (cahartin) [inline R/zchunk_L2522.ag_MACC_IRR_MGMT.R]: supply

**Comment by bpbond** [comment]
Holding on conflict resolution and approval by @cahartin .

**Pull request merged**